BLUESPAWN-win-client/src/util/permissions/permissions.cpp

```diff
@@ -133,35 +135,92 @@ namespace Permissions{
+	SecurityDescriptor SecurityDescriptor::CreateDACL(DWORD dwSize) {
 		return SecurityDescriptor(dwSize, SecurityDescriptor::SecurityDataType::DACL);
 	}
 
-	SecurityDescriptor SecurityDescriptor::CreateSACL(DWORD dwSize){
+	SecurityDescriptor SecurityDescriptor::CreateSACL(DWORD dwSize) {
 		return SecurityDescriptor(dwSize, SecurityDescriptor::SecurityDataType::SACL);
 	}
 
-	PACL SecurityDescriptor::GetDACL() const{ return this->dacl; }
-	PACL SecurityDescriptor::GetSACL() const{ return this->sacl; }
-	PSID SecurityDescriptor::GetUserSID() const{ return this->lpUserSID; }
-	PSID SecurityDescriptor::GetGroupSID() const{ return this->lpGroupSID; }
+	PACL SecurityDescriptor::GetDACL() const { return this->dacl; }
+	PACL SecurityDescriptor::GetSACL() const { return this->sacl; }
+	PSID SecurityDescriptor::GetUserSID() const { return this->lpUserSID; }
+	PSID SecurityDescriptor::GetGroupSID() const { return this->lpGroupSID; }
+
+	LsaHandleWrapper::LsaHandleWrapper(LSA_HANDLE handle) :
+		GenericWrapper(handle, std::function<void(LSA_HANDLE)>(SafeCloseLsaHandle), nullptr) {}
+
+	LsaHandleWrapper::LsaHandleWrapper(LSA_HANDLE handle, std::function<void(LSA_HANDLE)> fSafeClose) : 
+		GenericWrapper(handle, std::function<void(LSA_HANDLE)>(fSafeClose), nullptr) {}
+
+	bool Owner::bPolicyInitialized{ false };
+	LsaHandleWrapper Owner::lPolicyHandle{ nullptr };
+	const std::vector<std::wstring> Owner::vSuperUserPrivs{ SE_DEBUG_NAME, SE_IMPERSONATE_NAME, SE_TCB_NAME, SE_LOAD_DRIVER_NAME,
+			SE_ASSIGNPRIMARYTOKEN_NAME, SE_TAKE_OWNERSHIP_NAME };
+
+	void LsaHandleWrapper::SafeCloseLsaHandle(LSA_HANDLE handle) {
+		LsaClose(handle);
+	}
+
+	void Owner::InitializePolicy() {
+		LSA_HANDLE lTempPolicyHandle;
+		LSA_OBJECT_ATTRIBUTES lObjectAttr{};
+		HRESULT hr = LsaNtStatusToWinError(LsaOpenPolicy(nullptr, &lObjectAttr, GENERIC_ALL, &lTempPolicyHandle));
+		if (hr != ERROR_SUCCESS) {
+			bPolicyInitialized = false;
+			LOG_ERROR("Couldn't open policy handle. (Error:" << hr << ")");
+			SetLastError(hr);
+		}
+		else {
+			lPolicyHandle = { lTempPolicyHandle, std::function<void(LSA_HANDLE)>(Owner::DeinitializePolicy) };
+			bPolicyInitialized = true;
+		}
+	}
+
+	void Owner::DeinitializePolicy(LSA_HANDLE handle) {
+		LsaClose(handle);
+		bPolicyInitialized = false;
+	}
+
+	LSA_UNICODE_STRING Owner::WStringToLsaUnicodeString(IN const std::wstring& str) {
+		LSA_UNICODE_STRING lsaWStr{};
+		DWORD len = 0;
+
+		len = str.length();
+		AllocationWrapper cstr{ new WCHAR[len + 1], len + 1, AllocationWrapper::CPP_ARRAY_ALLOC };
+		memcpy(cstr, str.c_str(), (len + 1) * sizeof(WCHAR));
+		lsaWStr.Buffer = cstr.GetAsPointer<wchar_t>();
+		lsaWStr.Length = (USHORT)((len) * sizeof(WCHAR));
+		lsaWStr.MaximumLength = (USHORT)((len + 1) * sizeof(WCHAR));
+		return lsaWStr;
+	}
+
+	std::wstring Owner::LsaUnicodeStringToWString(IN const LSA_UNICODE_STRING& str) {
+		std::wstring toRet;
+		DWORD len = 0;
+
+		len = str.Length;
```

Comment:
Codacy found an issue: [Variable 'len' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5628317)